### ISSUE TITLE
Good first issues in rustworkx

### ISSUE THREAD

**Issue by mtreinish** [qiskit-advocate/qamp-spring-23#25]
### Description

[rustworkx](https://github.com/Qiskit/rustworkx) is a high performance general purpose graph library for python written in Rust to take advantage of the performance and safety that Rust provides. It was originally built as a replacement for Qiskit's previous networkx usage (hence the name) but is designed to provide a high performance general purpose graph library for any python application. The project was originally started to build a faster directed graph to use as the underlying data structure for the DAG at the center of qiskit-terra's transpiler, but it has since grown to cover all the graph usage both in Qiskit and other applications. We use rustworkx at the core of many of Qiskit's algorithms and functions (including the transpiler) when a graph representation is needed.

This project is to work on improving rustworkx and to help maintain the project. The primary activities around this project are going to be contributing to rustworkx and working on open issues. The full list of good first issues in rustworkx can be found [here](https://github.com/Qiskit/retworkx/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).

### Deliverables

The project is to work through good first issues on rustworkx. So ideally pull requests and reviews of others pull requests on rustworkx are the deliverable for this project. However, as it is a fairly open ended project to get involved in the development of rustworkx the exact deliverables aren't set in stone. In previous iterations mentees have helped opened pull requests to other projects using rustworkx along with other related tasks

### Mentors details

- Mentor 1
    - Name: Matthew Treinish
    - GitHub ID: @mtreinish
    - What they do: Qiskit Core Developer

### Number of mentees

3

### Type of mentees

Mentees

    Required:
        Some experience programming
        Desire to contribute to an open source project
    Nice to have:
        Experience in Python
        Experience in Rust (this is not expected and can be learned during the mentorship)
        Experience with graph theory

**Comment by AxlAlm** [comment]
Hi! 

I am very interested :) 

I work mostly with python but been trying to get more into rust!

**Comment by HitanshuGedam** [comment]
Hi!
I am interested in this project!
I do have some experience in Python and some in Rust (still learning) but no experience with graph theory(not until now!)
I'd be glad to work and contribute towards this project.

**Comment by YuhaoT** [comment]
Hi!

I am interested in this project as well~
I have experience in Python and currently learning Rust (reading the book) and would like to get some project experience in Rust.
I don't have any experience with graph theory but would like to learn more about it!

**Comment by beretests** [comment]
Hi! 

I'm interested in this project. I have some experience in Python. I have no experience in Rust or graph theory but I am also a quick learner and eager to learn.
